cinder/scheduler/filters/shard_filter.py

```diff
@@ -172,8 +164,10 @@ def _filter_by_k8s_cluster(self, backends, filter_properties):
         project_id = vol_props.get('project_id', None)
         metadata = vol_props.get('metadata', {})
 
+        is_vmware = any(self._is_vmware(b) for b in backends)
```

Comment:
That greatly depends on the order of filters, so we would need to filter first by `volume_backend_name`... I am not sure if that is done.

Reply:
yes, skip the query if the backend is not vmware based.....which currently only exists in qa-de-1.

Reply:
Currently, the filter is applied if *any* of the backends passed to the filter are vmware based.

Since the filter is almost right at the beginning, it will get all the backends in the AZ. We have in all regions VMWare based volumes, so the query will always apply, and filter out hosts by "shard", even if it is for another volume-type.

I would say, we need first to filter out the hosts by the `volume_backend_name` and `storage_protocol`, because that is how we target the vmware cinder-hosts or netapp cinder-hosts.  I think, that is done in the `CapabilitiesFilter`, but I am not sure about that.

If that is the case and we move the `CapabilitiesFilter` before the `ShardFilter`, then the code can stand as it is, as it won't be applied if we have a volume for a different hypervisor, because they will be filtered out before.

Reply:
Yes I believe that the CapabilitiesFilter would filter based on volume_backend_name by volume_type either through the default volume type (in our case vmware) or what the user specified as part of the request.

Either way standard_hdd or vmware would be allowed for non qa-de-1 regions.  


At this point it kinda makes sense to create a separate filter just for filtering by k8s clusters, since the shard filter is now filtering by 2 criteria instead of just by shard.